# Pandas

In [27]:
import pandas as pd
import numpy as np

df = pd.read_csv('diamonds.csv')

df.head()

carat      cut color clarity  depth  table  price     x     y     z
0   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75

In [21]:
df.tail()

carat        cut color clarity  depth  table  price     x     y     z
53935   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75  5.76  3.50
53936   0.72       Good     D     SI1   63.1   55.0   2757  5.69  5.75  3.61
53937   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66  5.68  3.56
53938   0.86    Premium     H     SI2   61.0   58.0   2757  6.15  6.12  3.74
53939   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83  5.87  3.64

In [22]:
df.sample(5)

carat        cut color clarity  depth  table  price     x     y     z
25527   2.02    Premium     I     SI2   59.4   59.0  14341  8.25  8.21  4.89
3997    0.76  Very Good     E     VS2   63.4   58.0   3517  5.76  5.82  3.67
11334   1.00       Good     E     SI1   62.6   61.0   4989  6.29  6.32  3.95
40317   0.40      Ideal     E     VS1   61.7   55.0   1125  4.76  4.73  2.93
19477   1.05    Premium     F     VS1   62.3   55.0   8114  6.57  6.49  4.07

In [36]:
df = pd.read_csv('/home/thuanc177/Documents/Git/Data-Analytic/patients.csv')

In [42]:
df.head()

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 503 entries, 0 to 502
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   patient_id    503 non-null    int64  
 1   assigned_sex  503 non-null    object 
 2   given_name    503 non-null    object 
 3   surname       503 non-null    object 
 4   address       491 non-null    object 
 5   city          491 non-null    object 
 6   state         491 non-null    object 
 7   zip_code      491 non-null    float64
 8   country       491 non-null    object 
 9   contact       491 non-null    object 
 10  birthdate     503 non-null    object 
 11  weight        503 non-null    float64
 12  height        503 non-null    int64  
 13  bmi           503 non-null    float64
dtypes: float64(3), int64(2), object(9)
memory usage: 55.1+ KB


# Nhận xét file patients.csv

Một số cột có giá trị rỗng (không có giá trị được ghi vào) như cột `địa chỉ` và `thông tin liên lạc`
Còn lại các cột liên quan đến *bệnh nhân*, giá trị đo đạc như **BMI**, **height**, **weight** đều được ghi lại đầy đủ

## Thông tin thêm

**Kiểu dữ liệu**: float64(3), int64(2), object(9)
**Bộ nhớ tiêu thụ**: 55.1+ KB

Thông tin height có kiểu dữ liệu int64 là không cần thiết khi int8 đã có thể biểu diễn 256 cm
Thay đổi này sẽ giúp giảm kích thước

In [43]:
df.describe()


patient_id      zip_code      weight      height         bmi
count  503.000000    491.000000  503.000000  503.000000  503.000000
mean   252.000000  49084.118126  173.434990   66.634195   27.483897
std    145.347859  30265.807442   33.916741    4.411297    5.276438
min      1.000000   1002.000000   48.800000   27.000000   17.100000
25%    126.500000  21920.500000  149.300000   63.000000   23.300000
50%    252.000000  48057.000000  175.300000   67.000000   27.200000
75%    377.500000  75679.000000  199.500000   70.000000   31.750000
max    503.000000  99701.000000  255.900000   79.000000   37.700000

# Chú thích

count là số hàng trong cột
std là độ lệch chuẩn
mean là trung bình cộng của cột
